### PR TITLE
[PD 3168] implement customerid control

### PR DESCRIPTION
**Pull request by koonsies**
## What's this PR do?

Destroys everything you love.

Just kidding. But it does make using CBO a little less convenient.

This PR adds CustomerID handling/enforcement to the CBO module. Rather than using a hardcoded CustomerID to display CBOs and such, the CustomerID will now be mapped to the currently selected company (via Toolbar/user status).

## Where should the reviewer start?
NOTE: This step may no longer be necessary. If there are any issues with accessing the CBO module with your currently selected company, follow these steps.

Start by running this script in your Django shell. Be sure to put your email in the proper place.

```
from myjobs.models import AppAccess, User
from seo.models import Company

de = Company.objects.get(name='AAR Corporation')
aa = AppAccess.objects.get(name='cbo') # Or any other appaccess
de.app_access.add(aa)

usr = User.objects.get(email="ENTER YOUR EMAIL")
dea = de.role_set.objects.all()[0] # grabs admin role
usr.roles.add(dea)
```

Once this is done, go to secure.my.jobs, select AAR as your company and access CBO.

## How should this be manually tested?

Well.. See above. You can also try selecting a company that doesn't have a CustomerID yet (like DirectEmployers) and see the pretty error it gives.

## What's Next

This PR also sets the stage for the opt in/opt out controls for CBO. This PR does not rely on that control, however, and can be merged without it.

### CODE REVIEW

**Comment by koonsies** [comment]
Adding DNM until an issue is resolved with the getCustomerID API

File: gulp/src/sass/cbo/cbo_main_styles.scss [hunk 1]
```diff
@@ -953,3 +957,92 @@ $large-desktops: 1200px;
...
+    .toggle-main-container {
+      display: inline-block;
+      position: relative;
+      text-align: left;
+      width: 60px;
+      height: 30px;
+      background-color: #484850;
+      overflow: hidden;
+      box-shadow: inset 0 10px 25px rgba(0,0,0, .5), 0px 1px 2px #fff;
+      -webkit-border-radius: 20px;
```
Comment (aljordan82) [inline gulp/src/sass/cbo/cbo_main_styles.scss]: Do we need the prefixes?

File: gulp/src/sass/cbo/cbo_main_styles.scss [hunk 2]
```diff
@@ -953,3 +957,92 @@ $large-desktops: 1200px;
...
+      -moz-border-radius: 20px;
+      border-radius: 20px;
+      display: inline-block;
+      width: 30px;
+      text-align: center;
+      font: bold 10px/28px Arial,Sans-Serif;
+      color: $white;
+      text-shadow: 0 -1px 0 rgba(0,0,0,0.7);
+      -webkit-transition: margin-left 0.2s ease-in-out;
```
Comment (aljordan82) [inline gulp/src/sass/cbo/cbo_main_styles.scss]: Can you replace this with a mixin?
`@mixin transition($transition...) {
    -moz-transition:   $transition;
    -o-transition:   $transition;
    -webkit-transition: $transition;
    transition:         $transition;
}`

File: gulp/src/sass/cbo/cbo_main_styles.scss (hunk 1 above)
Comment (mwarnsleyDE) [inline gulp/src/sass/cbo/cbo_main_styles.scss]: @aljordan82 yes we need prefixes for the other browsers

File: gulp/src/sass/cbo/cbo_main_styles.scss (hunk 2 above)
Comment (mwarnsleyDE) [inline gulp/src/sass/cbo/cbo_main_styles.scss]: @aljordan82 changed these to all mixins

**Pull request merged**